### PR TITLE
Add Chapel as a supported third-party language

### LINKED ISSUE #2800: [Help me] Add Chapel highlighting support
Hi highlight.js developers and experts —

I'm interested in adding support for the Chapel language (https://chapel-lang.org) to highlight.js, and managed to throw together a very simple (but better than nothing) highlighter on my laptop today that is building and rendering properly for me, and I'm looking for helps at next steps.  As a disclaimer, this is my first experience with node.js.

At this point I have two main questions:

* At the high level:  Reading the various contributing docs, I've ended up confused about whether the best practice is to: 
  - file a PR adding `chapel.js` to the main `highlight.js/highlight.js` repo...
    - ...under `src/languages/` (which is where most languages seem to be?)
    - ...or under `extra/...` (which is where `3RD_PARTY_QUICK_START.md` suggests it should be)
  - request a new Chapel-specific repository under `highlight.js` (which `3RD_PARTY_QUICK_START.md` alludes to?)
  - host it on our repo
  where my main goals is to have our Discourse site support Chapel highlighting and a secondary goal would be to have StackOverflow use it as well.

* At the low level: Though I've been able to do do a Chapel-specific build using `node tools/build.js -n chapel` and got reasonable output using `developer.html`, I've been unable to get a scripted all-languages build and test working using the instructions in the various docs, even from a clean copy of the repo that does not contain my changes.  E.g., from a copy of the tree in sync with the master/main branch (SHA = 1982c8caaf595bea16953fd0574ac2b152b2363d), I get:

```bash
$ node ./tools/build.js -t node
Starting build.
Finished build.
Writing styles.
Writing package.json.
/Users/bradc/highlight.js/tools/build_node.js:14
    let require = `require('./languages/${lang.name}')`;
                                               ^

TypeError: Cannot read property 'name' of undefined
    at /Users/bradc/highlight.js/tools/build_node.js:14:48
    at Array.map (<anonymous>)
    at buildNodeIndex (/Users/bradc/highlight.js/tools/build_node.js:13:34)
    at Object.buildNode [as build] (/Users/bradc/highlight.js/tools/build_node.js:97:9)
    at async doTarget (/Users/bradc/highlight.js/tools/build.js:87:3)
```

and a similar result when I use `npm run build_and_test`

Thanks for any advice with either of these issues.

### PR DESCRIPTION
**Pull request by bradcray**
Resolves #2800 

### Changes

This adds an entry for the Chapel language, giving `chapel` and `chpl` as its two aliases, and pointing to our repo at https://github.com/chapel-lang/highlightjs-chapel.  
### Checklist
- [x] Added markup tests, or they don't apply here because they're in our repo
- [x] Updated the changelog at `CHANGES.md`
- [ ] ~Added myself to `AUTHORS.txt`, under Contributors~ (wasn't certain whether adding a third-party package warranted being added to AUTHORS.txt and @joshgoebel didn't seem certain either)

[Note: I haven't done the last two items because it's not clear to me whether they apply to something as minor as adding a new third-party language to the list.  Please let me know]

### CODE REVIEW

**Comment by joshgoebel** [comment]
I dunno about authors (I won't stop ya, but I've never really thought about it), but you can definitely add an entry to the changelog if you'd like!

**Comment by bradcray** [comment]
@joshgoebel: I added a CHANGES.md entry as suggested.  I skipped the AUTHORS.md update, but can do it if you prefer.  Let me know if there are any other changes required before this is merged (Travis testing is still pending as I write this).

[Also, in other news, I successfully got our Discourse site using Chapel highlighting even without this being merged which I consider a major win]

File: CHANGES.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 ## Version 10.4.0 (work in process)
 
+New languages:
+
+- add(chapel) Added initial third-party Chapel language support (#2806) [Brad Chamberlain][]
```
Comment (joshgoebel) [inline CHANGES.md:5]:
```suggestion
- Added Chapel grammar to SUPPORTED_LANGUAGES (#2806) [Brad Chamberlain][]
```

**Review by joshgoebel — APPROVED** [review]

**Comment by bradcray** [comment]
Thanks Josh!

**Pull request merged**